src/components/organisms/Badges/Badges.tsx

```diff
@@ -1,64 +1,81 @@
-import React, { useMemo } from "react";
+import React, {
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
 import { User } from "@bugsnag/js";
 
 import { Link } from "react-router-dom";
-import { useFirestoreConnect } from "react-redux-firebase";
-
-import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
-import { useSelector } from "hooks/useSelector";
+import { useFirestore } from "react-redux-firebase";
 
 import { PLAYA_VENUE_NAME } from "settings";
 
 import { CampRoomData } from "types/CampRoomData";
 import { isVenueWithRooms } from "types/CampVenue";
-import { AnyVenue } from "types/Firestore";
+import { AnyVenue, UserVisit } from "types/Firestore";
 import { Venue } from "types/Venue";
 
 import { WithId } from "utils/id";
-import {
-  orderedVenuesSelector,
-  userModalVisitsSelector,
-} from "utils/selectors";
 import { venueInsideUrl, venuePreviewUrl } from "utils/url";
 import { notEmpty } from "utils/types";
 
 export const Badges: React.FC<{
   user: WithId<User>;
   currentVenue: WithId<Venue>;
 }> = ({ user, currentVenue }) => {
-  useFirestoreConnect([
-    {
-      collection: "users",
-      doc: user.id,
-      subcollections: [{ collection: "visits" }],
-      storeAs: "userModalVisits",
-    },
-  ]);
-  useFirestoreConnect("venues");
-  const visits = useSelector(userModalVisitsSelector);
-  const venues = useSelector(orderedVenuesSelector);
-
-  const { relatedVenues } = useConnectRelatedVenues({
-    venueId: currentVenue.id,
-    withEvents: false,
-  });
-  const relatedVenueNames = relatedVenues.map((venue) => venue.name);
+  const [visits, setVisits] = useState<WithId<UserVisit>[]>([]);
+  const [venues, setVenues] = useState<WithId<AnyVenue>[]>([]);
+
+  const imgRef: RefObject<HTMLImageElement[]> = useRef([]);
+
+  const firestore = useFirestore();
+
+  const fetchAllVenues = useCallback(async () => {
+    const venuesSnapshot = await firestore.collection("venues").get();
```

Comment:
Currently we seem to be loading all venues from the entire DB here, which we then `.map` a littler further down to get an array of venue names.

Do we need to fetch this? What is it being used for? If we do, can we make it only fetch the venues we need from our visits?

Reply:
Venues is being used for filtering of the visits that exist. Otherwise badges for possibly removed venues will appear.

Reply:
So can we get our badges, and then in batches of 10/similar, lookup whether the venue exists? That would save us from pulling every venue to the frontend, to then filter it down to only a few things.